test/connect.js

```diff
@@ -361,3 +364,48 @@ test('reject clients with wrong protocol name', function (t) {
     broker.on('closed', t.end.bind(t))
   })
 })
+
+// websocket-stream based connections
+test('websocket clients have access to the request object', function (t) {
+  t.plan(3)
+
+  var broker = aedes()
+  var server = http.createServer()
+  ws.createServer({
+    server: server
+  }, broker.handle)
+
+  server.listen(4883, function (err) {
+    t.error(err, 'no error')
+  })
+
+  broker.on('client', function (client) {
+    if (client.req) {
+      t.pass('client request object present')
+      if (client.req.headers) {
+        t.equal('sample', client.req.headers['x-test-protocol'])
+        finish()
+      }
+    } else {
+      t.fail('no request object present')
+    }
+  })
+
+  var client = mqtt.connect('ws://localhost:4883', {
+    wsOptions: {
+      headers: {
+        'X-Test-Protocol': 'sample'
+      }
+    }
+  })
+
+  var timer = setTimeout(finish, 1000)
```

Comment:
Could we eliminate the `setTimeout` ?

Reply:
If you feel it's necessary. It's copied from the same pattern in https://github.com/mcollina/aedes/blob/master/test/not-blocking.js#L18 as a clean up for failing tests.